### ISSUE TITLE
Create a centralized UI for multiple Argocd instances

### ISSUE THREAD

**Issue by chakch** [argoproj/argo-cd#11498]
# Summary

When we deploy multiple Argocd instances, It's difficult to follow what happens in all cluster. We need to login to each UI.
Having a centralized Argocd to deploy in all cluster, may lead to high privilege.

# Motivation

Having an centralized UI can make user management easier by providing a core deployment for each cluster and unified user management. 

# Proposal

api-server should support multiple repo-server.

**Comment by crenshaw-dev** [comment]
> api-server should support multiple repo-server

repo-server just renders manifests. I think what you're describing will require a much bigger architectural change.

> Having a centralized Argocd to deploy in all clusters, may lead to high privilege

How would that be avoided by having a centralized interface? What would you imagine the central UI giving you the ability to do/see?

**Comment by chakch** [comment]
> How would that be avoided by having a centralized interface? What would you imagine the central UI giving you the ability to do/see?
Each Argocd will deploy in-cluster and having a unique interface to see all Argocd instances

**Comment by payxn** [comment]
Hello,
Does anyone has a solution for this ? The only way I found is to create our own dashboard ...

**Comment by sreejith-opsmx** [comment]
@chakch @payxn Have you considered using an external tool for this?

**Comment by payxn** [comment]
@sreejith-opsmx Yes, but it was either too complicated either not free. We will build our own dashboard for the moment ...

**Comment by louisblin** [comment]
I feel this feature would let us get the best of both worlds in the deployment topology discussion*: do you run one Argo CD managing N clusters (unified UI but single point of failure) or does each cluster gets its own Argo CD (better isolation but many UIs)?

As our number of apps grows, we are starting to favor the latter solution as we try to improve our resilience to cluster / regional outages. More instances also reduces the blast radius of upgrades and helps with scalability. However, it does mean we need to give up our unified UI. It would be great if we could deploy Argo CD core in each cluster and run a single UI giving an overview of the plant.

\* Discussed in this [blog post](https://codefresh.io/blog/scaling-argo-cd-securely-in-2023/) from @daniel-codefresh

**Comment by zaphinath** [comment]
Due to security constraints with our company we have had to have multiple argocds. Some are parents of the others. We would love to have a way to unify the view of all of them.

**Comment by FuyuanChu** [comment]
Hello, Does anyone have a solution for this?  Currently, we ship the K8S clusters to our users with ArgoCD baked in. But there are too many hostnames of ArgoCD to remember and log in.

We hope there will be a unified UI that leverages a greater experience for using ArgoCD UI.

I think there are different levels of `unified`

1. A basic unification of the authentication. Users can only need to log in to the unified UI and switch the views to display the applications of different ArgoCD instances. In this case, most logics are still in the ArgoCD instances.

2. Advanced unification of RBAC. Besides the authentication, admin can set RBAC policies for the unified UI on resources. In this case, most logics are still in the ArgoCD instances, but the permissions are more fine-grained than the previous one.

3. Full functional unified UI. A replacement for the default ArgoCD UI that can manage resources on all clusters. In this case, the should be an independent control plane to rule the world of ArgoCD.

**Comment by crenshaw-dev** [comment]
afaik all existing options are paid. I believe some folks w/ OCM built a POC "pull-model" central Argo CD instance by having the central instance display read-only copies of the remote Applications. Not sure the state of that effort today. https://github.com/argoproj/argo-cd/issues/10908

**Comment by mtahaahmed** [comment]
Would love to get this feature

**Comment by gnunn1** [comment]
Does the [Argo CD Agent](https://github.com/argoproj-labs/argocd-agent) address this requirement? It basically distributes the application-controller among multiple clusters but maintains a central UI and API on a control plane cluster providing a centralized UI experience.
